lib/src/rules/prefer_enums.dart

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+import '../analyzer.dart';
+import '../ast.dart';
+import '../util/dart_type_utilities.dart';
+
+const _desc = r'Prefer enums over classes that behave like enums.';
```

Comment:
Wordsmithing here and below most welcome.  (I consider this text to be a placeholder at best.)

Reply:
Yeah.  I like `use_enums` a bit more.  Updated.  Thanks!